src/Abstractions/NexusMods.Abstractions.Library.NexusMods/NexusModsFileMetadata.cs

```diff
@@ -0,0 +1,35 @@
+using JetBrains.Annotations;
+using NexusMods.Abstractions.NexusWebApi.Types;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+
+namespace NexusMods.Abstractions.Library.NexusModsLibrary;
+
+/// <summary>
+/// Represents a remote file on Nexus Mods.
+/// </summary>
+[PublicAPI]
+public partial class NexusModsFileMetadata : IModelDefinition
+{
+    private const string Namespace = "NexusMods.Library.NexusModsFileMetadata";
+
+    /// <summary>
+    /// The ID of the file.
+    /// </summary>
+    public static readonly FileIdAttribute FileId = new(Namespace, nameof(FileId)) { IsIndexed = true };
+
+    /// <summary>
+    /// The name of the file.
+    /// </summary>
+    public static readonly StringAttribute Name = new(Namespace, nameof(Name));
+
+    /// <summary>
+    /// The version of the file.
+    /// </summary>
+    public static readonly StringAttribute Version = new(Namespace, nameof(Version));
+
+    /// <summary>
+    /// Reference to the mod page of the file.
+    /// </summary>
+    public static readonly ReferenceAttribute<NexusModsModPageMetadata> ModPage = new(Namespace, nameof(ModPage));
```

Comment:
No `GameDomain` on the file metadata? 
I know it is on the ModPage but maybe the duplication isn't a bad thing.
The game domain or id is available on the nxm link and file endpoint I'm pretty sure, so you would expect to find it on the file as well.